pkg/connector/tickets.go

```diff
@@ -0,0 +1,307 @@
+	l := ctxzap.Extract(ctx)
+	l.Info("Creating ticket", zap.Any("ticket", ticket))
+
+	payload, err := ln.createIssuePayloadFromTicket(ctx, ticket, schema)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	issue, err := ln.client.CreateIssue(ctx, *payload)
+	if err != nil {
+		return nil, nil, fmt.Errorf("baton-linear: failed to create issue: %w", err)
+	}
+
+	ticketResp := ticketFromIssue(issue)
+	return ticketResp, nil, nil
+}
+
+func (ln *Linear) GetTicketSchema(ctx context.Context, schemaID string) (*v2.TicketSchema, annotations.Annotations, error) {
+	teams, _, _, _, err := ln.client.ListTeamWorkflowStates(ctx, linear.GetTeamsVars{TeamIDs: []string{schemaID}, First: 2})
+	if err != nil {
+		return nil, nil, fmt.Errorf("baton-linear: failed to list team workflow states: %w", err)
+	}
+	if len(teams) != 1 {
+		return nil, nil, fmt.Errorf("baton-linear: expected 1 team, got %d", len(teams))
+	}
+	customFields, err := ln.getCustomFields(ctx)
+	if err != nil {
+		return nil, nil, fmt.Errorf("baton-linear: failed to list custom fields: %w", err)
+	}
+	return ticketSchemaFromTeam(teams[0], customFields), nil, nil
+}
+
+// ListTicketSchemas lists all the ticket schemas for Linear Issues.
+//
+// Linear Issues currently vary only by Workflow State per Team.
+func (ln *Linear) ListTicketSchemas(ctx context.Context, p *pagination.Token) ([]*v2.TicketSchema, string, annotations.Annotations, error) {
+	var annotations annotations.Annotations
+	bag, err := parsePageToken(p.Token, &v2.ResourceId{ResourceType: resourceTypeTeam.Id})
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	teams, nextToken, _, rlData, err := ln.client.ListTeamWorkflowStates(ctx, linear.GetTeamsVars{After: bag.PageToken(), First: resourcePageSize})
+	annotations.WithRateLimiting(rlData)
+	if err != nil {
+		return nil, "", annotations, fmt.Errorf("baton-linear: failed to list teams: %w", err)
+	}
+
+	pageToken, err := bag.NextToken(nextToken)
+	if err != nil {
+		return nil, "", annotations, err
+	}
+
+	customFields, err := ln.getCustomFields(ctx)
+	if err != nil {
+		return nil, "", annotations, fmt.Errorf("baton-linear: failed to list custom fields: %w", err)
+	}
+
+	var ret []*v2.TicketSchema
+	for _, team := range teams {
+		ret = append(ret, ticketSchemaFromTeam(team, customFields))
+	}
+
+	return ret, pageToken, annotations, nil
+}
+
+func ticketSchemaFromTeam(team linear.Team, customFields map[string]*v2.TicketCustomField) *v2.TicketSchema {
+	var statuses []*v2.TicketStatus
+
+	sort.Slice(team.States.Nodes, func(i, j int) bool {
+		return team.States.Nodes[i].Position < team.States.Nodes[j].Position
+	})
+
+	for _, state := range team.States.Nodes {
+		statuses = append(statuses, &v2.TicketStatus{
+			Id:          state.ID,
+			DisplayName: state.Name,
+		})
+	}
+
+	return &v2.TicketSchema{
+		Id:          team.ID,
+		DisplayName: team.Name,
+		Statuses:    statuses,
+		Types: []*v2.TicketType{
+			{
+				Id:          "issue",
+				DisplayName: "Issue",
+			},
+		},
```

Comment:
nit: we actually don't use this field anymore, but it doesn't do anything if you set it so this is fine

Reply:
Removed (b795ef4).

Happy to reduce any unnecessary noise in the codebase.